### PR TITLE
Make SessionState members private

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change

Exposing members publicly is a very strong and broad API commitment, which makes it very difficult to make non-breaking changes, as it is very hard to reason about how the type is being used.

# What changes are included in this PR?

Makes the members of `SessionState` private

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/planner.rs [hunk 1]
```diff
@@ -1773,7 +1771,7 @@ impl DefaultPhysicalPlanner {
     where
         F: FnMut(&dyn ExecutionPlan, &dyn PhysicalOptimizerRule),
     {
-        let optimizers = &session_state.physical_optimizers;
+        let optimizers = session_state.physical_optimizers();
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/planner.rs:1774]: This is quite a peculiar API, I'm not sure why the physical planner would be calling the optimizers and not SessionState itself

Comment (alamb) [reply datafusion/core/src/physical_plan/planner.rs:1774]: it may predate when the list of optimizers was on the session state 🤔

**Review by alamb — APPROVED** [review]
LGTM -- thank you @tustvold

**Comment by tustvold** [comment]
I plan to merge this tomorrow morning unless I hear anything further

File: datafusion/core/src/physical_plan/planner.rs (hunk 1 above)
Comment (tustvold) [reply datafusion/core/src/physical_plan/planner.rs:1774]: Ok - I will add it to my list

**Review by jackwener — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Pull request merged**